app/scripts/controllers/sponsormodal.js

```diff
@@ -1,20 +1,63 @@
 'use strict';
 
 angular.module('bulbsCmsApp')
-  .controller('SponsormodalCtrl', function ($scope, ContentFactory, article, Campaign) {
+  .controller('SponsormodalCtrl', function ($scope, ContentFactory, article, Campaign,
+                                            // NOTE: temp depdendencies will be moved along with searchCampaigns/loadCampaigns
+                                            _, $q, $http, TunicConfig) {
     $scope.article = article;
 
-    if ($scope.article.campaign) {
-      $scope.campaign = Campaign.$find($scope.article.campaign);
+    $scope.tunicCampaignIdMapping = {};
+
+    // --------------------------- TODO: MOVE ME SOMEWHERE ELSE---------------------------------------
+    var $loadTunicCampaign = function (tunicCampaignId) {
+      if (_.isNumber(tunicCampaignId)) {
+        return $http.get(TunicConfig.buildBackendApiUrl('campaign/' + tunicCampaignId + '/')).then(function (result) {
+          return result.data;
+        });
+      }
+      return $q.reject();
+    };
+
+    var $searchCampaigns = function (params) {
+      return $http.get(TunicConfig.buildBackendApiUrl('campaign/'), {
+        params: params,
+      }).then(function (response) {
+        return response.data.results;
+      });
+    };
+    // --------------------------- END TODO ---------------------------------------
+
+    if ($scope.article.tunic_campaign_id) {
+      $loadTunicCampaign($scope.article.tunic_campaign_id).then(function (campaign) {
+        $scope.tunicCampaignId = campaign.id;
```

Comment:
Do I even need to set this, or can i just reference $scope.article.tunic_campaign_id inside the HTML element?